ground/src/main/java/com/google/android/ground/persistence/remote/firebase/FirebaseFirestoreProvider.kt

```diff
@@ -27,17 +27,22 @@ import timber.log.Timber
 @Singleton
 class FirebaseFirestoreProvider @Inject constructor(settings: FirebaseFirestoreSettings) :
   AsyncSingletonProvider<FirebaseFirestore>({
-    FirebaseFirestore.getInstance().also {
-      if (it.firestoreSettings == settings) {
-        FirebaseFirestore.setLoggingEnabled(Config.FIRESTORE_LOGGING_ENABLED)
-        return@also
-      }
-      try {
-        it.firestoreSettings = settings
-      } catch (e: IllegalStateException) {
-        // Logging added for #2377.
-        Timber.e(e, "Singleton provider likely initialized multiple times")
-      }
-      FirebaseFirestore.setLoggingEnabled(Config.FIRESTORE_LOGGING_ENABLED)
-    }
+    FirebaseFirestore.getInstance().apply { setFirestoreSettings(this, settings) }
   })
+
+private fun setFirestoreSettings(
+  firestore: FirebaseFirestore,
+  settings: FirebaseFirestoreSettings,
+) {
+  if (firestore.firestoreSettings == settings) {
+    FirebaseFirestore.setLoggingEnabled(Config.FIRESTORE_LOGGING_ENABLED)
```

Comment:
This can be moved out of the if block and removed from bottom as well

Reply:
Done